core/services/privacy_policy/PrivacyPolicyManager.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace EventEspresso\core\services\privacy_policy;
+
+use EventEspresso\core\exceptions\InvalidClassException;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidEntityException;
+use EventEspresso\core\exceptions\InvalidFilePathException;
+use EventEspresso\core\exceptions\InvalidIdentifierException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use EventEspresso\core\services\collections\CollectionDetails;
+use EventEspresso\core\services\collections\CollectionInterface;
+use EventEspresso\core\services\collections\CollectionLoader;
+use WP_Post;
+
+defined('EVENT_ESPRESSO_VERSION') || exit('No direct script access allowed');
+
+
+
+/**
+ * Class PrivacyPolicyManager
+ * Manages setting up the hooks to add the EE core and add-ons' privacy policies
+ *
+ * @package        Event Espresso
+ * @author         Mike Nelson
+ * @since          $VID:$
+ */
+class PrivacyPolicyManager
+{
+
+    public function __construct()
+    {
+        if (function_exists('wp_add_privacy_policy_content')) {
+            add_action('edit_form_after_title', array($this, 'maybeAddPrivacyPolicy'), 9);
+        }
+    }
+
+
+
+    /**
+     * For all the registered `PrivacyPolicyInterface`s, add their privacy policy content
+     *
+     * @param WP_Post $post
+     */
+    public function addPrivacyPolicy($post)
+    {
+        if (! ($post instanceof \WP_Post)) {
+            return;
+        }
+        $policy_page_id = (int)get_option('wp_page_for_privacy_policy');
+        if (! $policy_page_id || $policy_page_id != $post->ID) {
+            return;
+        }
+        //load all the privacy policy stuff
+        //add post policy text
+        foreach ($this->loadPrivacyPolicyCollection() as $privacy_policy) {
+            wp_add_privacy_policy_content($privacy_policy->getName(), $privacy_policy->getContent());
+        }
+    }
+
+
+
+    /**
+     * @return CollectionInterface|EventEspresso\core\services\privacy_policy\PrivacyPolicyInterface[]
+     * @throws InvalidIdentifierException
+     * @throws InvalidInterfaceException
+     * @throws InvalidFilePathException
+     * @throws InvalidEntityException
+     * @throws InvalidDataTypeException
+     * @throws InvalidClassException
+     */
+    protected function loadPrivacyPolicyCollection()
+    {
+        $loader = new CollectionLoader(
+            new CollectionDetails(
+            // collection name
+                'privacy_policies',
+                // collection interface
+                'EventEspresso\core\services\privacy_policy\PrivacyPolicyInterface',
+                // FQCNs for classes to add (all classes within that namespace will be loaded)
+                apply_filters(
+                    'FHEE__EventEspresso_core_domain_services_admin_privacy_policy_Manager__privacy_policies',
```

Comment:
should update filter named after classname change

Reply:
good catch